File: tip_calculator.js
// tip_calculator.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function askQuestion(query) {
    return new Promise(resolve => rl.question(query, resolve));
}

async function getNumber(prompt, isInt = false) {
    while (true) {
        const input = await askQuestion(prompt);
        const val = isInt ? parseInt(input) : parseFloat(input);
        if (!isNaN(val) && val >= 0 && (isInt ? Number.isInteger(val) && val > 0 : true)) {
            return val;
        }
        console.log('Invalid input. Please enter a valid number.');
    }
}

async function main() {
    console.log('=== Tip Calculator ===');
    const bill = await getNumber('Enter bill amount: ');
    const tipPercent = await getNumber('Enter tip percentage (e.g., 15 for 15%): ');
    const people = await getNumber('Enter number of people: ', true);

    const tip = bill * (tipPercent / 100);
    const total = bill + tip;
    const tipPerPerson = tip / people;
    const totalPerPerson = total / people;

    console.log('\n--- Result ---');
    console.log(`Bill: $${bill.toFixed(2)}`);
    console.log(`Tip (${tipPercent.toFixed(0)}%): $${tip.toFixed(2)}`);
    console.log(`Total: $${total.toFixed(2)}`);
    console.log(`Split among ${people} people:`);
    console.log(`Tip per person: $${tipPerPerson.toFixed(2)}`);
    console.log(`Total per person: $${totalPerPerson.toFixed(2)}`);

    rl.close();
}

main().catch(console.error);

File: tip_calculator.go
// tip_calculator.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strconv"
    "strings"
)

func main() {
    reader := bufio.NewReader(os.Stdin)

    fmt.Print("Enter bill amount: ")
    billStr, _ := reader.ReadString('\n')
    bill, err := strconv.ParseFloat(strings.TrimSpace(billStr), 64)
    if err != nil {
        fmt.Println("Invalid bill amount.")
        return
    }

    fmt.Print("Enter tip percentage (e.g., 15 for 15%): ")
    tipStr, _ := reader.ReadString('\n')
    tipPercent, err := strconv.ParseFloat(strings.TrimSpace(tipStr), 64)
    if err != nil {
        fmt.Println("Invalid tip percentage.")
        return
    }

    fmt.Print("Enter number of people: ")
    peopleStr, _ := reader.ReadString('\n')
    people, err := strconv.Atoi(strings.TrimSpace(peopleStr))
    if err != nil || people <= 0 {
        fmt.Println("Invalid number of people.")
        return
    }

    tipAmount := bill * (tipPercent / 100)
    total := bill + tipAmount
    tipPerPerson := tipAmount / float64(people)
    totalPerPerson := total / float64(people)

    fmt.Printf("\n--- Result ---\n")
    fmt.Printf("Bill: $%.2f\n", bill)
    fmt.Printf("Tip (%.0f%%): $%.2f\n", tipPercent, tipAmount)
    fmt.Printf("Total: $%.2f\n", total)
    fmt.Printf("Split among %d people:\n", people)
    fmt.Printf("Tip per person: $%.2f\n", tipPerPerson)
    fmt.Printf("Total per person: $%.2f\n", totalPerPerson)
}
